include/deal.II/grid/cell_status.h

```diff
@@ -0,0 +1,53 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_cell_status_h
+#define dealii_cell_status_h
+
+#include <deal.II/base/config.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * Used to inform functions in derived classes how the cell with the given
+ * cell_iterator is going to change. Note that this may me different than
+ * the refine_flag() and coarsen_flag() in the cell_iterator in parallel
+ * calculations because of refinement constraints that this machine does not
+ * see.
+ */
+enum CellStatus
+{
+  /**
+   * The cell will not be refined or coarsened and might or might not move
+   * to a different processor.
+   */
+  CELL_PERSIST,
+  /**
+   * The cell will be or was refined.
+   */
+  CELL_REFINE,
+  /**
+   * The children of this cell will be or were coarsened into this cell.
+   */
+  CELL_COARSEN,
+  /**
+   * Invalid status. Will not occur for the user.
+   */
+  CELL_INVALID
+};
```

Comment:
I recognize that this will require a good bit of additional work, but I think it should be done anyway:

When you have an `enum` in the global namespace, as you have here, it injects the names into the global namespace as well. This is undesirable. A better approach would be to make it an `enum class`, which keeps the names in a separate namespace named after the `enum class` itself (here, `CellStatus`).

Since you're already making incompatible changes, perhaps using names that are more according to our usual naming scheme would be useful. I think these names might be `CellStatus::cell_will_persist`, `CellStatus::cell_will_be_refined`, `CellStatus::children_will_be_coarsened`, and `CellStatus::invalid`?

Reply:
Done, thank you!